pandas/tests/io/test_sql.py

```diff
@@ -475,6 +514,28 @@ def sqlite_conn(sqlite_engine):
         yield conn
 
 
+@pytest.fixture
+def sqlite_adbc_conn(iris_path):
+    if pa_version_under8p0:
+        pytest.skip("ADBC requires pyarrow >= 8.0.0")
+    pytest.importorskip("adbc_driver_sqlite")
+    import adbc_driver_manager as mgr
+    from adbc_driver_sqlite import dbapi
+
+    with tm.ensure_clean() as name:
+        uri = f"file:{name}"
+        with dbapi.connect(uri) as conn:
+            try:
+                conn.adbc_get_table_schema("iris")
+            except mgr.InternalError:  # note arrow-adbc issue 1022
+                conn.rollback()
+                create_and_load_iris(conn, iris_path, "sqlite")
```

Comment:
@jorisvandenbossche I think your comment was lost in the diff, but here is where I am loading iris. Its a little different than what you suggested, but I figured the advantage was less code duplication by keeping this all in one fixture. Not a hard design decision for me so happy to break this out if that's what you prefer

Reply:
This is certainly better. My patch was just a quick copy/past hack to get some test running to check if it was passing or not ;)